### PR TITLE
LC-332 - update reactivation process

### PR DESCRIPTION
**Pull request by m24murray**
_Also needed is the following PR on identity-service: https://github.com/Civil-Service-Human-Resources/identity-service/pull/163_

New service and data classes for updates to reactivation. When a user submits a reactivation, a reactivation object will be created and an email will be sent to the user to confirm the reactivation. This allows reactivation requests to be sent through the agency token validation if neccessary.

- new sent request reactivation flow for reactivating a user
- new gov notify template id for reactivation
- updates to templates and controllers for update identity
- new custom exceptions

### CODE REVIEW

File: src/test/java/uk/gov/cshr/controller/IdentityControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        doNothing().when(identityService).updateActive(UID);
+
+        mockMvc.perform(
+                post("/identities/active")
+                        .with(csrf())
+                        .accept(APPLICATION_JSON).param("uid", UID))
+                .andDo(print())
+                .andExpect(model().attributeDoesNotExist("status"))
+                .andExpect(status().is3xxRedirection());
+    }
```
Comment (brdgeth) [inline src/test/java/uk/gov/cshr/controller/IdentityControllerTest.java:62]: Theres unit tests missing

File: src/test/java/uk/gov/cshr/service/security/IdentityServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                csrsService,
+                notificationService,
+                messageService,
+                resetService,
+                tokenService,
+                identityReactivationService);
+    }
+
+    @Test
+    public void shouldSetActiveToFalseIfActive() {
```
Comment (brdgeth) [inline src/test/java/uk/gov/cshr/service/security/IdentityServiceTest.java:80]: change to shouldSetActiveToFalseIfActiveIsTrue()

File: src/test/java/uk/gov/cshr/service/security/IdentityServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    private IdentityReactivationService identityReactivationService;
+
+    private int deactivationMonths = 10;
+
+    private int notificationMonths = 20;
+
+    private int deletionMonths = 36;
+
+    private IdentityService identityService;
+
```
Comment (brdgeth) [inline src/test/java/uk/gov/cshr/service/security/IdentityServiceTest.java:62]: unit tests missing

Comment (m24murray) [reply src/test/java/uk/gov/cshr/service/security/IdentityServiceTest.java:62]: Added additional tests to cover any new journeys

**Comment by m24murray** [comment]
> Unit testing is a little light on the service, could do with up'ing

Added tests to cover new journeys

**Pull request merged**